File: dotnet/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example4/Car.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgeDesignPattern.Example4;

public class Car : Vehicle
{
	public Car(Workshop workshop1, Workshop workshop2) : base(workshop1, workshop2)
	{
	}

	public override void Manufacture()
	{
		Console.Write("Car");
		_workShop1.Work();
		_workShop2.Work();
	}
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example4/src/Car.ts
import { IWorkshop } from './IWorkshop';
import { Vehicle } from './Vehicle';

export class Car extends Vehicle {
  constructor(w1: IWorkshop, w2: IWorkshop) { super(w1, w2); }
  manufacture(): void {
    console.log(`Car${this.workshop1.work()}${this.workshop2.work()}`);
  }
}
